src/main/java/clients/symphony/api/DatafeedClientV2.java

```diff
@@ -0,0 +1,145 @@
+package clients.symphony.api;
+
+import clients.SymBotClient;
+import clients.symphony.api.constants.AgentConstants;
+import configuration.SymConfig;
+import exceptions.SymClientException;
+import exceptions.UnauthorizedException;
+import model.DatafeedEvent;
+import model.DatafeedV2EventList;
+import model.StringId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.core.GenericType;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.HashMap;
+import java.util.List;
+
+import static javax.ws.rs.core.Response.Status.Family.SUCCESSFUL;
+
+public class DatafeedClientV2 extends APIClient implements DatafeedClient {
+
+    private final Logger logger = LoggerFactory.getLogger(DatafeedClientV2.class);
+    private SymBotClient botClient;
+    private SymConfig config;
+
+    public DatafeedClientV2(SymBotClient client) {
+        this.botClient = client;
+        this.config = client.getConfig();
+    }
+
+    @Override
+    public String createDatafeed() throws SymClientException {
+        Invocation.Builder builder = botClient.getAgentClient()
+                .target(config.getAgentUrl())
+                .path(AgentConstants.CREATEDATAFEEDV2)
+                .request(MediaType.APPLICATION_JSON)
+                .header("sessionToken", botClient.getSymAuth().getSessionToken())
+                .header("keyManagerToken", botClient.getSymAuth().getKmToken());
+
+        logger.info("Create datafeed for bot {}..", botClient.getBotUsername());
+        try (Response response = builder.post(null)) {
+            if (response.getStatusInfo().getFamily() != SUCCESSFUL) {
+                try {
+                    handleError(response, botClient);
```

Comment:
My 2 cents: I don't like that logic because it is complex and it hides the fact that `handleError` authenticates again if we have an unauthorized response. And logic is duplicated in `readDatafeed` method.
We will address that later in the v2, I think.

Reply:
Yes indeed! Because now I want to do it quickly and make it align with other client, I reused the legacy retry strategy in v1. But clearly we need to improve it in v2!

Reply:
You are right guys, let's keep it as it is in V1 and make error handling way more better in V2 👍